cmd/cosmos-sdk-cli/cmd/init.go

```diff
@@ -0,0 +1,81 @@
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"go/build"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/version"
+	"github.com/gobuffalo/packr"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	rootCmd.AddCommand(initCmd)
+}
+
+func resolveProjectPath(remoteProjectPath string) string {
+	gopath := os.Getenv("GOPATH")
+	if gopath == "" {
+		gopath = build.Default.GOPATH
+		// Use $HOME/go
+	}
+	return gopath + string(os.PathSeparator) + "src" + string(os.PathSeparator) + remoteProjectPath
+}
+
+var initCmd = &cobra.Command{
+	Use:   "init AwesomeProjectName",
+	Short: "Initialize your new cosmos zone",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) < 1 {
+			return errors.New("Project name is required")
+		}
+		projectName := args[0]
+		capitalizedProjectName := strings.Title(projectName)
+		shortProjectName := strings.ToLower(projectName)
+		reader := bufio.NewReader(os.Stdin)
+		fmt.Println("Thank you for using cosmos-zone tool.")
+		fmt.Println("You are only a few steps away from creating your brand new blockchain project on Cosmos.")
+		fmt.Print("We will ask you a few more questions to guide you through this process\n\n")
+		fmt.Print("To configure this project we need a remote project path. If you are unsure you can leave this empty. ")
+		fmt.Print("Remote project path is usually something like github.com/your_user_name/project_name\n")
+		fmt.Print("Enter remote project path: ")
+		remoteProjectPath, _ := reader.ReadString('\n')
+		remoteProjectPath = strings.ToLower(strings.TrimSpace(remoteProjectPath))
+		if remoteProjectPath == "" {
+			remoteProjectPath = strings.ToLower(shortProjectName)
+		}
+		projectPath := resolveProjectPath(remoteProjectPath)
+		fmt.Print("configuring the project in " + projectPath + "\n\n")
+		time.Sleep(2 * time.Second)
+		box := packr.NewBox("../template")
```

Comment:
We're generally attempting to mimimize the use of external dependancies in the SDK, I'm pretty sure we can iterate through the files and replace their contents without the use of `packr`.

Reply:
Packr is a library to bundle the template into the cosmos-sdk-cli binary. It provides an easy to use functionality to box the files/folders at the time of producing the binary and then use those files later when the binary is run. I think packr handles relative paths very well. If we have to implement, we have to check the GOPATH, and then correctly point to the basecoin folder in cosmos-sdk and then copy the contents of the file. This would work, but there are other advantages with using packr. 
1. Due to iterative nature of cosmos-sdk, if there is any incorrect link to the template folder, it could be caught at the time of build, rather than at the time of running the commands by user, which is a bad user experience for people using the SDK.
2. Handling relative paths is very easy using packr. This might need careful coding to make sure we are choosing the right path to templates if we are to implement this ourselves.

Please think about it and let me know if you still want to remove the packr dependency. I can remove it, not a big deal, but there are surely some benefits to using packr especially for cosmos-sdk-cli.

Reply:
cool - I'm pretty happy leaving with packr removed for now - but I agree it shouldn't really be a problem to include external deps for helper binaries like this, we just need to make sure we got the SDK on lockdown - but with time sure we can figure out a way to include it. Let's leave for now though